facet-format-postcard: unconditional miette dependency breaks WASM builds

## Summary

After updating to the latest facet (commit 9096c8c6, which includes #1475), WASM builds fail because `facet-format-postcard` has unconditional `impl miette::Diagnostic` blocks that don't compile when miette isn't available.

## Error

```
error[E0433]: failed to resolve: use of unresolved module or unlinked crate `miette`
  --> facet-format-postcard/src/error.rs:27:6
   |
27 | impl miette::Diagnostic for PostcardError {
   |      ^^^^^^ use of unresolved module or unlinked crate `miette`

error[E0433]: failed to resolve: use of unresolved module or unlinked crate `miette`
  --> facet-format-postcard/src/error.rs:97:6
   |
97 | impl miette::Diagnostic for SerializeError {}
   |      ^^^^^^ use of unresolved module or unlinked crate `miette`
```

## Reproduction

```toml
# Cargo.toml - WASM build without JIT
[target.'cfg(target_arch = "wasm32")'.dependencies]
facet-format-postcard = { git = "https://github.com/facet-rs/facet", branch = "main", default-features = false }
```

```bash
cargo check --target wasm32-unknown-unknown
```

## Expected Behavior

The `miette::Diagnostic` impls should be behind a feature gate (e.g., `#[cfg(feature = "miette")]`) so that WASM builds without the full feature set can still compile.

## Context

This is blocking rapace's WASM support. We use `default-features = false` for WASM because cranelift JIT doesn't support WASM targets.

--- a/facet-format-postcard/Cargo.toml
+++ b/facet-format-postcard/Cargo.toml
@@ -19,7 +19,7 @@ rustdoc-args = ["--html-in-header", "arborium-header.html"]
 facet-core = { path = "../facet-core", version = "0.34.0", default-features = false }
 facet-format = { path = "../facet-format", version = "0.34.0" }
 facet-path = { path = "../facet-path", version = "0.34.0", default-features = false, features = ["alloc"] }
-facet-reflect = { path = "../facet-reflect", version = "0.34.0", default-features = false, features = ["miette"] }
+facet-reflect = { path = "../facet-reflect", version = "0.34.0", default-features = false }
 miette = { workspace = true, optional = true }
 
 # Axum integration (optional)
@@ -64,7 +64,7 @@ harness = false
 default = ["std", "pretty-errors"]
 std = ["alloc", "facet-core/std", "facet-reflect/std", "facet-path/std"]
 alloc = ["facet-core/alloc", "facet-reflect/alloc", "facet-path/alloc"]
-pretty-errors = ["facet-path/pretty", "dep:miette"]
+pretty-errors = ["facet-path/pretty", "facet-reflect/miette", "dep:miette"]
 jit = ["facet-format/jit"]
 
 # Axum HTTP integration

--- a/facet-format-postcard/src/error.rs
+++ b/facet-format-postcard/src/error.rs
@@ -24,6 +24,7 @@ impl fmt::Display for PostcardError {
 
 impl std::error::Error for PostcardError {}
 
+#[cfg(feature = "pretty-errors")]
 impl miette::Diagnostic for PostcardError {
     fn code<'a>(&'a self) -> Option<Box<dyn fmt::Display + 'a>> {
         Some(Box::new(format!("postcard::error::{}", self.code)))
@@ -94,4 +95,5 @@ impl fmt::Display for SerializeError {
 
 impl std::error::Error for SerializeError {}
 
+#[cfg(feature = "pretty-errors")]
 impl miette::Diagnostic for SerializeError {}

